lightning/src/routing/gossip.rs

```diff
@@ -1483,6 +1512,19 @@ impl<L: Deref> NetworkGraph<L> where L::Target: Logger {
 			}
 		}
 
+		#[cfg(feature = "std")]
+		{
+			let removed_nodes = self.removed_nodes.lock().unwrap();
+			let removed_channels = self.removed_channels.lock().unwrap();
```

Comment:
Let's document the locking order. May also want to be consistent and lock channels before nodes, if that makes sense.